src/System.Private.CoreLib/src/System/IO/FileLoadException.CoreCLR.cs

```diff
@@ -21,12 +20,12 @@ private FileLoadException(string fileName, string fusionLog, int hResult)
             _message = FormatFileLoadExceptionMessage(FileName, HResult);
         }
 
-        internal static string FormatFileLoadExceptionMessage(string fileName, int hResult)
+        internal static string FormatFileLoadExceptionMessage(string? fileName, int hResult)
         {
-            string format = null;
+            string? format = null;
```

Comment:
I wonder whether `out` would work as well as `ref` here, so you don't need to make it nullable. @stephentoub ?

Reply:
It would require changing GetStringHandleOnStack to accept out instead of ref, but other than any issues associated with that (e.g. its implementation currently uses Unsafe.AsPointer which takes a ref rather than out), I'm not aware of anything fundamental that would prevent that, and all callers would appear to work with out instead of ref. @jkotas?

Reply:
Yes, there should not be anything fundamental that would prevent this.

- The annotation would be technically incorrect, like in other places. The local is still null when the `GetStringHandleOnStack` method returns. It becomes non-null only after the QCall that the handle is passed into returns.
- I am not sure whether all current uses of `GetStringHandleOnStack` return non-null (most of them do).
- There is a sister method `GetObjectHandleOnStack` that is used to pass objects in; and that is used to return null in some cases. Having one shape for `GetStringHandleOnStack` and different shape for `GetObjectHandleOnStack` would look odd. If we really wanted to have non-null returning string variant, I would give it a different name, like `GetReturnStringHandleOnStack`.

This feels like a general problem with wrappers, helpers or state objects. The nullability does not propagate through these and one has to compensate for it manually.

Reply:
> This feels like a general problem with wrappers, helpers or state objects.

Yes, the annotations aren't sufficient to express what needs to be expressed.  Various attributes are being pondered to help (https://github.com/dotnet/roslyn/issues/26761).

Reply:
Added TODO comment.